### PR TITLE
modules associated with chaperone

### PR DESCRIPTION
**Pull request by jdatx**

### CODE REVIEW

File: vcenter_vrops_config.py [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+    import requests
+    import inspect
+    import logging
+    IMPORTS = True
+except ImportError:
+    IMPORTS = False
+
+## Logging
+LOG = logging.getLogger(__name__)
+handler = logging.FileHandler('/var/log/chaperone/vcenter_vrops_config.log')
```
Comment (tompscanlan) [inline vcenter_vrops_config.py]: Log file ought to be a parameter

File: vcenter_vrops_deploy.py [hunk 2]
```diff
@@ -113,190 +95,344 @@
...
-        connected = (vm.runtime.connectionState == 'connected')
-
-        if connected:
-            powered_on = (vm.runtime.powerState == 'poweredOn')
-
-            if powered_on:
-                return True
+## Logging
+LOG = logging.getLogger(__name__)
+handler = logging.FileHandler('/var/log/chaperone/vcenter_vrops_deploy.log')
```
Comment (tompscanlan) [inline vcenter_vrops_deploy.py]: log ought to be a param

File: vcenter_vrops_deploy.py [hunk 3]
```diff
@@ -18,12 +18,13 @@
...
 module: vcenter_vrops_deploy
-Short_description: Deploys (creates), Deletes vROps ova to vcenter cluster
+Short_description: Deploys (creates), Deletes vROPs ova to vcenter cluster
 description:
-    Deploys (creates), Deletes vROps ova to vcenter cluster. Module will wait for vm to
-    power on.
+    Deploys, Deletes vROPs ova to vcenter cluster. Module will wait for vm to
+    power on and "pings" the vROPs api before exiting if not failed.
 requirements:
     - pyvmomi 6
```
Comment (tompscanlan) [inline vcenter_vrops_deploy.py]: Looks like there is now a requirement for ovftool, add that here

File: vcenter_vrops_deploy.py [hunk 4]
```diff
@@ -113,190 +95,344 @@
...
+
+        self.module.exit_json(changed=changed, result=result, msg=msg)
+
+    def get_vm(self, vm_name):
+        vm = find_vm_by_name(self.si, vm_name)
+        return vm
+
+    def deploy_ova(self):
+
+        ovftool_exec = '{}/ovftool'.format(self.module.params['ovftool_path'])
```
Comment (tompscanlan) [inline vcenter_vrops_deploy.py]: note requirement for ovftool

File: vcenter_vrops_deploy.py [hunk 5]
```diff
@@ -113,190 +95,344 @@
...
+            resp = requests.get(url=url, verify=False, headers=header)
+        except requests.exceptions.ConnectionError:
             return False
 
+        log("Status code: {}".format(resp.status_code))
+        return resp.status_code
+
+    def wait_for_api(self, sleep_time=15):
+        status_poll_count = 0
+        while status_poll_count < 30:
```
Comment (tompscanlan) [inline vcenter_vrops_deploy.py]: poll count ought to be a param, with a safe default value

File: vio_cluster_deploy.py [hunk 6]
```diff
@@ -55,10 +54,23 @@
...
+        required: True
+        type: str
     state:
         description:
             - desired state of the vio cluster
-        choices: present, absent 
+        choices: present, absent
+
+
+requirements:
```
Comment (tompscanlan) [inline vio_cluster_deploy.py]: line out of place?

File: vio_cluster_deploy.py [hunk 7]
```diff
@@ -80,12 +93,13 @@
...
     import requests
     import inspect
+    import collections
     IMPORTS = True
 except ImportError:
     IMPORTS = False
 
 LOG = logging.getLogger(__name__)
-handler = logging.FileHandler('/tmp/vio_cluster_log.log')
+handler = logging.FileHandler('/var/log/chaperone/vio_cluster_deploy.log')
```
Comment (tompscanlan) [inline vio_cluster_deploy.py]: param for log?

File: vio_oms_deploy.py [hunk 8]
```diff
@@ -207,51 +252,58 @@ def find_virtual_machine(content, searched_vm_name):
...
 
+    if module.params['state'] != 'absent':
+        vapp_data = get_vapp_data()
+        vapp_id = vapp_data['id']
+        vapp_name = vapp_data['name']
 
-def state_create_vapp(module):
+    module.exit_json(changed=False, object_id=vapp_id, object_name=vapp_name, msg="EXIT UNCHANED")
 
+def state_create_vapp(module):
```
Comment (tompscanlan) [inline vio_oms_deploy.py]: Note requirement for ovftool

**Comment by jdatx** [comment]
Just wanted to see if there were anymore changes needed, if not can this be merged?

**Pull request merged**